verifier/src/qe_identity.rs

```diff
@@ -80,20 +142,68 @@ impl<'a> TryFrom<&SignedQeIdentity<'a>> for QeIdentity {
     }
 }
 
-#[derive(Debug, Deserialize)]
+#[derive(Debug, Deserialize, Clone, PartialEq)]
 #[serde(rename_all = "camelCase")]
-struct TcbLevel {
+pub struct TcbLevel {
     tcb: Tcb,
     tcb_date: String,
     tcb_status: AdvisoryStatus,
     #[serde(rename = "advisoryIDs", default)]
     advisory_ids: Vec<String>,
 }
 
-#[derive(Debug, Deserialize)]
-struct Tcb {
+impl TcbLevel {
+    pub fn new<'a, I, E>(tcb: Tcb, tcb_status: AdvisoryStatus, advisory_ids: I) -> Self
+    where
+        I: IntoIterator<Item = &'a E>,
+        E: ToString + 'a + ?Sized,
+    {
+        Self {
+            tcb,
+            tcb_date: UNIX_TIME_STR.to_string(),
+            tcb_status,
+            advisory_ids: advisory_ids.into_iter().map(ToString::to_string).collect(),
+        }
+    }
+
+    pub fn isv_svn(&self) -> IsvSvn {
+        self.tcb.isv_svn.into()
+    }
+
+    pub fn status(&self) -> AdvisoryStatus {
+        self.tcb_status
+    }
+
+    pub fn advisory_ids(&self) -> &[String] {
+        &self.advisory_ids
+    }
+}
+
+// A default value that will not be verified to be correct.
+// It will be `AdviceStatus::OutOfDate` and have an `isv_svn` of 0.
+impl Default for TcbLevel {
```

Comment:
This felt a bit more ergonomic to have an invalid default TcbLevel, but the verification failure messages seem a bit odd, so perhaps this should go away.

Reply:
I've updated to use `Option<TcbLevel>` instead, this default has now been removed.

Sorry re-reading these comments, I realize I left quite a bit of context in my head and didn't write it out.